### ISSUE TITLE
[Spree Upgrade] Fix admin orders edit - capture payment (1 broken test)

### ISSUE THREAD

**Issue by luisramos0** [openfoodfoundation/openfoodnetwork#3113]
Fix:

```
  6) 
    As an administrator
    I want to manage orders
 capture payment from the orders index page
     Failure/Error: expect(page).to have_content "Payment Updated"
       expected to find text "Payment Updated" in "Logged In As : lynn_kuhlman@cormier.name Account Logout Store DASHBOARD ORDERS PRODUCTS REPORTS CONFIGURATION USERS ORDER CYCLES GROUPS ENTERPRISES CUSTOMERS ORDERS BULK ORDER MANAGEMENT SUBSCRIPTIONS Listing Orders NEW ORDER SEARCH DATE RANGE STATUS INVOICE NUMBER: EMAIL FIRST NAME BEGINS WITH LAST NAME BEGINS WITH ONLY SHOW COMPLETE ORDERS DISTRIBUTORS ORDER CYCLES FILTER RESULTS 15 per page 1 Results found. Viewing 1 to 1. DISTRIBUTOR ORDER DATE NUMBER STATE PAYMENT STATE SHIPMENT STATE CUSTOMER E-MAIL TOTAL Enterprise 31 November 20, 2018 R244243478 [MISSING \"EN.ORDER_STATE.COMPLETE\" TRANSLATION] [MISSING \"EN.PAYMENT_STATES.BALANCE_DUE\" TRANSLATION] [MISSING \"EN.SHIPMENT_STATES.PENDING\" TRANSLATION] anastasia.maggio@monahan.com $13.00 « FIRST PREVIOUS 1 NEXT LAST »"
     # ./spec/features/admin/orders_spec.rb:185:in `block (2 levels) in <top (required)>'
```

**Comment by luisramos0** [comment]
#3286 added the missing translation.
The the payment state is updated correctly but there’s some problem in the order updater because order.update! is being called but the order.payment_state is not updated from the order.payments state (the issue 3113 was updated with this info)

**Comment by mkllnk** [comment]
The problem seems to be that the order is not paid in full:

| Time                        | order.total | order.payment_state | payment.amount | payment.state | 
|--------------------------|---------------|----------------------------|-----------------------|-------------------|
|Before capturing       | 13             | balance_due              | 10                       | checkout        |
|During order update | 13             | balance_due              | 10                       | completed      |
|After order update    | 13             | balance_due              | 10                       | completed      |

The payment exists already and has the wrong value. The line item's price is 10. There is a shipping fee of 3 and a transaction fee of 0. So somehow the payment doesn't include the shipping fee.

**Comment by luisramos0** [comment]
awesome catch @mkllnk 
thanks a lot. I am not sure you were planning to fix it yourself or not, but I couldn't resist giving it a try now that you have found the problem. I think this is the fix #3299 Thanks!

**Comment by luisramos0** [comment]
when this issue is closed, we can close #3110 and #3124
and if the build is green, we can also close #2927
